libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -189,11 +189,16 @@ static void _networkReceiveTask( void * pArgument )
          * may only be called once (per its API doc). */
         connectionFlags = xEventGroupGetBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ) );
 
-        if( ( connectionFlags & _FLAG_CONNECTION_DESTROYED ) == _FLAG_CONNECTION_DESTROYED )
+        /* Break out of receive task loop if connection is closed or destroyed. */
+        if( ( connectionFlags & _FLAG_RECEIVE_TASK_CONNECTION_DESTROYED ) == _FLAG_RECEIVE_TASK_CONNECTION_DESTROYED )
         {
             destroyConnection = true;
             break;
         }
+        else if( ( connectionFlags & _FLAG_SHUTDOWN ) == _FLAG_SHUTDOWN )
+        {
+            break;
+        }
```

Comment:
This block seems redundant? Don't we already do the same thing inside the `do...while` or is there something I'm missing?

Reply:
This is a different case to detect the calling of `IotNetworkAfr_Close` (or specifically, the `.close` member) function from within the receive callback.
The earlier logic within the do-while loop detects the connection closure if it happens _before_ the receive callback is invoked.

Reply:
Okay, that makes sense. Need to handle both cases.